x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/node/label_node/label_node.tsx

```diff
@@ -0,0 +1,164 @@
+      >
+        <LabelNodeContainer data-test-subj={TEST_SUBJ_CONTAINER}>
+          {interactive && (
+            <LabelShapeOnHover
+              data-test-subj={TEST_SUBJ_HOVER_OUTLINE}
+              color={
+                analysis.isSingleAlert || analysis.isGroupOfAlerts
+                  ? euiTheme.colors.danger
+                  : euiTheme.colors.primary
+              }
+            />
+          )}
+          <LabelShape
+            data-test-subj={TEST_SUBJ_SHAPE}
+            backgroundColor={backgroundColor}
+            borderColor={borderColor}
+            textAlign="center"
+            shadow={shadow}
+          >
+            <div
+              css={css`
+                display: flex;
+                align-items: center;
+                justify-content: space-between;
+                width: 100%;
+                gap: ${euiTheme.size.xs};
+              `}
+            >
+              <EuiText
+                color={textColor}
+                css={css`
+                  flex: 1;
+                  min-width: 0;
+                  text-overflow: ellipsis;
+                  font-weight: ${euiTheme.font.weight.semiBold};
+                  font-size: ${euiTheme.font.scale.xs * 10.5}px;
+                `}
+              >
+                <EuiTextTruncate truncation="end" text={text}>
+                  {(truncatedText) => truncatedText}
+                </EuiTextTruncate>
+              </EuiText>
+              <LabelNodeBadges analysis={analysis} />
+            </div>
+          </LabelShape>
+          {interactive && (
+            <>
+              <NodeButton
+                css={css`
+                  margin-top: -${ACTUAL_LABEL_HEIGHT}px;
+                `}
+                height={ACTUAL_LABEL_HEIGHT}
+                width={NODE_LABEL_WIDTH}
+                onClick={(e) => nodeClick?.(e, props)}
+              />
+              <NodeExpandButton
+                data-test-subj={TEST_SUBJ_EXPAND_BTN}
+                color={'primary'}
+                onClick={(e, unToggleCallback) => expandButtonClick?.(e, props, unToggleCallback)}
+                x={`${NODE_LABEL_WIDTH - 3}px`}
+                y={`${
+                  -ACTUAL_LABEL_HEIGHT +
+                  (ACTUAL_LABEL_HEIGHT - NodeExpandButton.ExpandButtonSize) / 2
+                }px`}
+              />
+            </>
+          )}
+          <Handle
+            data-test-subj={TEST_SUBJ_HANDLE}
+            type="target"
+            isConnectable={false}
+            position={Position.Left}
+            id="in"
+            style={HandleStyleOverride}
+          />
+          <Handle
+            data-test-subj={TEST_SUBJ_HANDLE}
+            type="source"
+            isConnectable={false}
+            position={Position.Right}
+            id="out"
+            style={HandleStyleOverride}
+          />
+        </LabelNodeContainer>
+      </EuiToolTip>
+      <LabelNodeDetails ips={ips} countryCodes={countryCodes} />
```

Comment:
nit, when you have a component that can result no rendering, I find it better not to render it at all.  (when no ips nor countryCodes)

It can save us from unexpected results